app/controllers/api/v1/sessions_controller.rb

```diff
@@ -41,20 +41,27 @@ def session_params
       end
 
       def sign_in(user)
+        user.generate_session_token!(extended_session: session_params[:extend_session])
+
         # Creates an extended_session cookie if extend_session is selected in sign in form.
         if session_params[:extend_session]
           cookies.encrypted[:_extended_session] = {
             value: {
-              user_id: user.id
+              session_token: user.session_token,
+              session_expiry: user.session_expiry
             },
             expires: 7.days,
-            httponly: true
+            httponly: true,
+            secure: true
           }
         end
-        session[:user_id] = user.id
+
+        session[:session_token] = user.session_token
+        session[:session_expiry] = user.session_expiry
```

Comment:
Setting the `session_expiry` (which may be not required as mentionned above) will cause the signin logic to differ/diverge from external accounts signin (which does set only the `session_digest`).
However, since the `session_expiry` could be omitted that does not have a critical implication except for enlarging cookies size and its implication (4KB limit + slower requests loading [it's not much for this project scope]).